### ISSUE TITLE
Debug weather override affects all save files

### ISSUE THREAD

**Issue by hexagonrecursion** [CleverRaven/Cataclysm-DDA#54133]
### Describe the bug

Setting weather via debug leaks into other save files. Only quitting to desktop removes the override.

~~I appear to be stuck in a permanent postal storm. The storm did not end when the message "The damage to reality fades.  Hopefully the scarring is minimal" was printed. Saving and loading did not fix it and neither did waiting for several days or driving to a new overmap.~~

### Steps to reproduce

1. Create a new world
2. Set weather via debug (e.g. portal storm)
3. Save and quit
4. Create a second world
5. See that the weather override has affected both worlds

~~Sorry, I don't know how to reproduce this. I was crafting. A portal storm started. I drove to a different location and then went to sleep in my car. While sleeping I saw a chat message: "The damage to reality fades.  Hopefully the scarring is minimal", but when I woke up the portal storm was still going.~~

### Expected behavior

I expected the weather override to only affect the world it was executed in just like all other debug options. I expected having two saves - one for debugging and one for regular play - to be a supported usecase.

~~I expected the portal storm to end when the message appeared.~~

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: :core-4.1-amd64:core-4.1-noarch; Distributor ID: Fedora; Description: Fedora release 34 (Thirty Four); Release: 34; Codename: ThirtyFour; 
- Game Version: 4d22129 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

### Additional context

[tmp2.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7829052/tmp2.zip)

* https://github.com/CleverRaven/Cataclysm-DDA/issues/52491 is similar, but *my* bug does not disappear no matter how many times I save and load.
* https://github.com/CleverRaven/Cataclysm-DDA/issues/53642 is similar, but *my* world was definitely created using a game version newer then https://github.com/CleverRaven/Cataclysm-DDA/pull/53435

**Comment by hexagonrecursion** [comment]
Note: I have fixed my save by editing it: triggering EOC_CANCEL_PORTAL_STORM again, but a less tech-savvy player might not know how to do this. Also: a player that does not know how the storms are supposed to work may assume this is intentional.

**Comment by hexagonrecursion** [comment]
Update: the storm did clear up on its own shortly after I exited to desktop and loaded the game again. It would appear that reloading the game w/o quitting to desktop does not fix this.

**Comment by Ramza13** [comment]
> Update: the storm did clear up on its own shortly after I exited to desktop and loaded the game again. It would appear that reloading the game w/o quitting to desktop does not fix this.

That is very interesting.  There have been a bunch of issues around this and weather changing and I have no idea what's causing them, its quite frustrating.  The fact that its possibly fixed by a quit but not a reload is interesting.  Thanks and any more details you notice are very much appreciated, this issue has been a real pain to find so far.

**Comment by PatrikLundell** [comment]
I suggest your spam filter is broken if you're subjected to a postal storm...

**Comment by hexagonrecursion** [comment]
I remembered something. This was my first portal storm. When it started I wanted to know how worried should I be - how deadly it is and whether I'll be safe inside a car - because the game has killed me by mechanics I did not even know existed way too many times. So I created a debugging world and set the weather to portal storm to test out the effects. Could it be that the debug weather override does not get reset when loading a different save?

Yep! This is what was causing the bug for me. I'll update the title and description.
